### PR TITLE
Adding Dev Container

### PR DESCRIPTION
**Pull request by Ellerbach**
This PR adds support for [dev containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
This will allow to have standard and always working build for all our targets without having to install anything on the Windows machine! It does as well bring Linux build support indirectly.

## Description

Dev Container is a concept of having a Docker container running and connected to VS Code. This allows an isolation of a development environment and the current host OS. In other words, this creates a Linux based docker image with all the tools installed and everything already setup. nothing has to be added.
The first time, the creation of the container takes some time but then there is nothing additonal to do and if you connect a second time, all is still there.

## Motivation and Context

A standard, always working environment without having to install anything on the Windows machine, so in other words to avoid polluting the machine with all the tool chains and their dependencies.

## How Has This Been Tested?

Every image that are in the cmake-variants-DEVCONTAINER.json has been tested. They all prepare properly and compile!

Known limitation: hex2dfu will not execute. The build will be marked as failed. But all the files to create the dfu are present. The hex2dfu will need to be fixed to support Linux environments. In the meantime, this manual step will be needed.

Documentation to be updated in a coming PR on the github.io documentation. The full process has been tested with community members as well and will be helpful to create the documentation.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/cmake-variants.json [hunk 1]
```diff
@@ -27,9 +27,9 @@
         "settings": {
           "BUILD_VERSION": "0.9.99.999",
           "CMAKE_TOOLCHAIN_FILE": "CMake/toolchain.xtensa-esp32-elf.cmake",
-          "TOOLCHAIN_PREFIX": "<absolute-path-to-the-esp32-toolchain-folder-mind-the-forward-slashes>",
```
Comment (josesimoes) [inline targets/FreeRTOS_ESP32/ESP32_WROOM_32/cmake-variants.json:30]: No need to replace these here. This is a template and these strings are to be replaced by the install script. If they are changed it won't work.

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/cmake-variants.json [hunk 2]
```diff
@@ -69,9 +69,9 @@
...
```
Comment (josesimoes) [inline targets/FreeRTOS_ESP32/ESP32_WROOM_32/cmake-variants.json:72]: Same here

**Review by josesimoes — CHANGES_REQUESTED** [review]

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/cmake-variants.json (hunk 1 above)
Comment (Ellerbach) [reply targets/FreeRTOS_ESP32/ESP32_WROOM_32/cmake-variants.json:30]: correct! missed that. Will adjust

**Review by josesimoes — APPROVED** [review]
LGTM and to the build pipeline as well! 😉 💯

**Pull request merged**